libr/anal/esil.c

```diff
@@ -1387,16 +1384,21 @@ static int esil_poke_n(RAnalEsil *esil, int bits) {
 		if (dst && r_anal_esil_get_parm (esil, dst, &addr)) {
 			int type = r_anal_esil_get_parm_type (esil, src);
 			if (type != R_ANAL_ESIL_PARM_INTERNAL) {
-				n.qword = n2.qword = 0;
-				r_anal_esil_mem_read (esil, addr, (ut8 *)&n, bytes);
-				r_mem_copyendian ((ut8 *)&n2, (ut8 *)&n, bytes, !esil->anal->big_endian);
-				esil->old = n2.qword;
+				r_anal_esil_mem_read (esil, addr, b, bytes);
+				if (!esil->anal->big_endian)
```

Comment:
using ternary operator will look better imho. also, use positive conditionals unless for error handling